server/routes/index.js

```diff
@@ -0,0 +1,93 @@
+import * as usersController from '../controllers/users';
+import * as groupsController from '../controllers/groups';
+import authenticate from '../middleware/authenticate';
+import isGroupMember from '../middleware/isGroupMember';
+import isValidUsername from '../middleware/isValidUsername';
+import isTaken from '../middleware/isTaken';
+
+module.exports = (app) => {
+  app.get('/api', (req, res) => res.status(200).send({
+    message: 'Welcome to the Postit API!',
+    info: 'This is a list of available end points',
+    signup: 'POST /api/user/signup',
+    signin: 'POST /api/user.signin',
+    getMe: 'GET /api/user/me',
+    myGroups: '/api/user/me/groups',
+    myMessages: '/api/user/me/messages',
+    logout: 'POST /api/user/logout',
+    createGroup: '/api/group',
+    addUserToGroup: '/api/group/:groupid/user',
+    groupUsers: '/api/group/:groupid/users',
+    sendMessage: '/api/group/:groupid/message',
+    getMessages: '/api/group/:groupid/messages'
+  }));
+
+  // User routes
+  app.post('/api/user/signup', isValidUsername, isTaken,
+   usersController.signup);
+
+  app.post('/api/user/signin', isValidUsername,
+    usersController.signin);
+
+  app.get('/api/user/me', authenticate, usersController.getMe);
+
+  // app.get('/api/user/token', authenticate, usersController.refreshToken);
+
+  app.get('/api/user/all', authenticate, usersController.getAllUsers);
+
+  app.get('/api/user/me/groups', authenticate, usersController.getMyGroups);
+
+  app.get('/api/user/me/messages', authenticate,
+  usersController.getMySentMessages);
+
+  app.patch('/api/user/me/password', authenticate,
+  usersController.changePassword);
+
+  // app.patch('/api/user/me/email', authenticate, usersController.changeEmail);
+
+  // app.post('/api/user/deactivate', authenticate, isValidUsername,
+  // usersController.deactivate);
+
+  // Group routes
+  app.post('/api/group', authenticate, groupsController.createGroup);
+
+  app.post('/api/group/:groupid/user', authenticate, isValidUsername,
+  isGroupMember, groupsController.addUserToGroup);
+
+  app.delete('/api/group/:groupid/user', authenticate, isValidUsername,
+  isGroupMember, groupsController.removeUserFromGroup);
+
+  app.post('/api/group/:groupid/leave', authenticate,
+  isGroupMember, groupsController.leaveGroup);
+
+  app.get('/api/group/:groupid/users', authenticate,
+  isGroupMember, groupsController.getGroupUsers);
+
+  app.post('/api/group/:groupid/message', authenticate,
```

Comment:
I see you are currently using express v4.x which provides you with a router instance to chain endpoint of different methods. Use this as a guide 

https://www.codementor.io/johnkingzy/api-routing-using-expressjs-router-4-0-9yohet6uu